tools/system_libs.py

```diff
@@ -28,6 +28,14 @@
                 'getsockopt.c', 'setsockopt.c', 'freeaddrinfo.c',
                 'in6addr_any.c', 'in6addr_loopback.c', 'accept4.c']
 
+# Experimental: Setting EMCC_USE_NINJA will cause system libraries to get built with ninja rather
+# than simple subprocesses.  The primary benefit here is that we get accurate dependency tracking.
+# This means we can avoid completely rebuilding a library and just rebuild based on what changed.
+#
+# Setting EMCC_USE_NINJA=2 means that ninja will automatically be run for each library needed at
+# link time.
```

Comment:
I don't understand the difference between 1 and 2 in this comment. Reading the code, it seems like in mode 2 we force rebuilding of all libraries, even ones already built?

Reply:
Yes kind of.  `2` means than whenever we need access to a given library we re-run the ninja file.. which will then do all the dependency checks as rebuild as needed.    This mean you can edit e.g. dlmalloc.c and then just run `emcc hello.c` and it will trigger the rebuilding of just that one file in that one library.

`1` will use ninja to build things but won't run the ninja files at all if that libraries already exist.. you don't get the "edit" + `emcc` + "everything just works" workflow.

This interface isn't final or ideal, its just something I cam up with for this experimental phase.